iOS/DuckDuckGo/AppServices/ReportingService.swift

```diff
@@ -60,34 +62,33 @@ final class ReportingService {
     private func onStatisticsLoaded() {
         fireAppLaunchPixel()
         reportAdAttribution()
+        reportWidgetUsageAsync()
         onboardingPixelReporter.fireEnqueuedPixelsIfNeeded()
     }
 
     private func fireAppLaunchPixel() {
-        WidgetCenter.shared.getCurrentConfigurations { result in
-            let paramKeys: [WidgetFamily: String] = [
-                .systemSmall: PixelParameters.widgetSmall,
-                .systemMedium: PixelParameters.widgetMedium,
-                .systemLarge: PixelParameters.widgetLarge
-            ]
-
-            switch result {
-            case .failure(let error):
-                Pixel.fire(pixel: .appLaunch, withAdditionalParameters: [
-                    PixelParameters.widgetError: "1",
-                    PixelParameters.widgetErrorCode: "\((error as NSError).code)",
-                    PixelParameters.widgetErrorDomain: (error as NSError).domain
-                ], includedParameters: [.appVersion, .atb])
-
-            case .success(let widgetInfo):
-                let params = widgetInfo.reduce([String: String]()) {
-                    var result = $0
-                    if let key = paramKeys[$1.family] {
-                        result[key] = "1"
+        Pixel.fire(pixel: .appLaunch, includedParameters: [.appVersion, .atb])
+    }
+
+    private func reportWidgetUsageAsync() {
+        guard featureFlagging.isFeatureOn(.widgetReporting) else { return }
+
+        Task {
```

Comment:
are we sure calling widget center from non-main thread is safe?

Reply:
API don't say its required and it seems to work.  Internally it could be using the main thread.  Concurrency version isn't available until iOS 18 so hence the callback approach still.